ui/components/src/Legend/ListLegendItem.tsx

```diff
@@ -30,6 +30,7 @@ export const ListLegendItem = React.memo(function ListLegendItem({ item }: ListL
         padding: 0,
         cursor: 'pointer',
       }}
+      data-testid={`legend-item-${item.color}`}
```

Comment:
This is temporary, I'm planning to change how item.id is generated when I make changes at the query level